rustls/src/suites.rs

```diff
@@ -117,6 +118,18 @@ impl SupportedCipherSuite {
                 .is_some(),
         }
     }
+
+    /// Return the list of `KeyExchangeAlgorithm`s supported by this cipher suite.
+    ///
+    /// TLS 1.3 cipher suites support both ECDHE and DHE key exchange, but TLS 1.2 suites
+    /// support one or the other.
+    pub(crate) fn key_exchange_algorithms(&self) -> &[KeyExchangeAlgorithm] {
+        match self {
+            #[cfg(feature = "tls12")]
+            Self::Tls12(tls12) => core::slice::from_ref(&tls12.kx),
+            Self::Tls13(_) => ALL_KEY_EXCHANGE_ALGORITHMS,
```

Comment:
Is it correct to return `ALL_KEY_EXCHANGE_ALGORITHMS` that contains `KeyExchangeAlgorithm::DHE` even if we have not add FFDHE support for TLS 1.3?

Reply:
From what I understand, TLS 1.3 suites support both ECDHE and DHE, that's why TLS 1.3 suites don't have the key exchange algorithm in their names.

Reply:
Got it!